src/material/form-field/resize-observer.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {Injectable, OnDestroy} from '@angular/core';
+
+class SingleBoxSharedResizeObserver {
+  private _observer?: ResizeObserver;
+  private _callbacks = new Map<Element, Set<(entry: ResizeObserverEntry) => void>>();
+
+  constructor(private _box: ResizeObserverBoxOptions) {}
+
+  observe(target: Element, callback: (entry: ResizeObserverEntry) => void): () => void {
```

Comment:
Instead of passing in a callback here, I think that we should return an observable and have it tear itself down in the `complete` callback. See how the `ContentObserver` does it.

Reply:
Done